### PR TITLE
Feature/ios 1553 scrollable trimmer

### PR DESCRIPTION
**Pull request by talsharon**

### CODE REVIEW

File: PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift [hunk 1]
```diff
@@ -11,117 +11,182 @@ import UIKit
...
+    let layout: UICollectionViewFlowLayout = UICollectionViewFlowLayout()
+    var maxOnscreenDuration: Double = 1800
+    fileprivate var thumbnailFrameAspectRatio: CGFloat?
+    fileprivate var duration: TimeInterval?
+    fileprivate var thumbnailTimes: [NSValue] = []
+    fileprivate var thumbnailSize: CGSize = CGSize.zero
+    fileprivate var contentWidth: CGFloat = 0
+    fileprivate var lastWidth: CGFloat?
+    fileprivate var lastContentOffset: CGFloat = 0
+    fileprivate var horizontalInset: CGFloat = 15
```
Comment (nadavhart) [inline PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift]: all of the above need to be private

File: PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift [hunk 2]
```diff
@@ -11,117 +11,182 @@ import UIKit
...
-        generateImages(at: thumbnailTimes, with: thumbnailSize, visibleThumnails: thumbnailCount)
+        let thumbnailCount =  getThumbnailCount(for: duration)
+        thumbnailTimes = getThumbnailTimes(for: duration, numberOfThumbnails: thumbnailCount)
+        self.collectionView.reloadData()
+        self.collectionView.performBatchUpdates(nil) { _ in
+            self.delegate?.didUpdateDimensions()
+        }
+    }
+    
+    private func getThumbnailCount(for duration: TimeInterval) -> Int {
```
Comment (nadavhart) [inline PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift]: In iOS it is common not to start functions or vars with "get". Just "thumbnailCount" is the best practice.

File: PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift [hunk 3]
```diff
@@ -11,117 +11,182 @@ import UIKit
...
+            return 0
+        }
+        let thumbnailSize = getThumbnailFrameSize(for: thumbnailFrameAspectRatio)
+        let thumbnailCount =  thumbnailSize.width > 0 ? Int(ceil(contentWidth / thumbnailSize.width)) : 0
+        return thumbnailCount
     }
 
-    private func getThumbnailFrameSize(for aspectRatio: CGFloat) -> CGSize? {
-        let height = frame.height
+    private func getThumbnailFrameSize(for aspectRatio: CGFloat) -> CGSize {
```
Comment (nadavhart) [inline PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift]: same as above

File: PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift [hunk 4]
```diff
@@ -132,37 +197,106 @@ class AssetVideoScrollView: UIScrollView {
...
     }
+}
 
-    private func generateImages(at times: [NSValue], with maximumSize: CGSize, visibleThumnails: Int) {
-        let scaledSize = CGSize(width: maximumSize.width * UIScreen.main.scale,
-                                height: maximumSize.height *  UIScreen.main.scale)
-        var count = 0
+extension AssetVideoScrollView {
+    
+    func getTime(from position: CGFloat) -> CMTime? {
```
Comment (nadavhart) [inline PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift]: same as above

File: PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift (hunk 1 above)
Comment (talsharon) [reply PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift]: compiled with swift 3.0 :(

File: PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift (hunk 2 above)
Comment (talsharon) [reply PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift]: used an existing func, but I'l change 👍

**Comment by talsharon** [comment]
@nadavhart

File: PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift (hunk 1 above)
Comment (nadavhart) [reply PryntTrimmerView/Classes/Parents/AssetVideoScrollView.swift]: Basa. Can you add an issue to the convert this one to latest Swift please?

**Pull request merged**